ios/Events/OnPageSelected.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+import React
+
+@objc public class RCTOnPageSelected: NSObject, RCTEvent {
+    private var position: NSNumber
+    @objc public var viewTag: NSNumber
+    @objc public var coalescingKey: UInt16
+    
+    @objc public var eventName: String {
+        return "onPageSelected"
+    }
+    
+    @objc public init(reactTag: NSNumber, position: NSNumber, coalescingKey: UInt16) {
+        self.viewTag = reactTag
+        self.position = position
+        self.coalescingKey = coalescingKey
+        super.init()
+    }
+    
+    @objc public func canCoalesce() -> Bool {
+        return false
+    }
+    
```

Comment:
If `canCoalesce` is false you don't need to have `coalescingKey` and `public func coalesce(with newEvent: RCTEvent) ->  RCTEvent`

Reference: https://github.com/facebook/react-native/blob/67bff8734f4b92fe399910eecad5b67511a749c1/packages/react-native/React/Base/RCTEventDispatcherProtocol.h#L51-L56